src/ptvsd/adapter/messages.py

```diff
@@ -183,12 +184,30 @@ def attach_request(self, request):
         options.port = int(request.arguments.get("port", options.port))
         _channels.connect_to_server(address=(options.host, options.port))
 
-        return self._configure()
+        return self._configure(request)
+
+    def _set_debugger_properties(self, args):
+        client_os_type = None
+        if 'WindowsClient' in args.get('debugOptions', []):
```

Comment:
Take a look at how "debugOptions" are handled in debuggee.py to add some validation here. If `debugOptions` is not a JSON array, this will crash the adapter immediately, rather than reporting request as invalid.

Reply:
Is this for the VS style debugOptions?